### PR TITLE
refactor chain accumulation and add tests

### PR DESCRIPTION
**Pull request by jeanphilippeD**
chain_accumulator & completed_events work together.
We will need to update them to handle signing BLS key, so extract them and add unit test.

Partially address https://github.com/maidsafe/routing/issues/1768

### CODE REVIEW

**Comment by jeanphilippeD** [comment]
Fixed review comments inline

File: src/chain/chain_accumulator.rs [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+        let proofs = self.chain_accumulator.remove(&event)?;
+
+        if !self.completed_events.insert(event.clone()) {
+            log_or_panic!(LogLevel::Warn, "Duplicate insert in completed events.");
+        }
+
+        Some((event, proofs))
+    }
+
+    pub fn uncomplete_events(&self) -> impl Iterator<Item = (&NetworkEvent, &ProofSet)> {
```
Comment (fizyk20) [inline src/chain/chain_accumulator.rs]: Should this method be called `incomplete_events`?

File: src/chain/chain_accumulator.rs [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    fn get_first_proof(proofs: &ProofSet) -> Proof {
+        let (pub_id, sig) = unwrap!(proofs.sigs.iter().next());
+
+        Proof {
+            pub_id: *pub_id,
+            sig: *sig,
+        }
+    }
+
+    fn uncomplete_events(acc: &ChainAccumulator) -> Vec<(NetworkEvent, ProofSet)> {
```
Comment (fizyk20) [inline src/chain/chain_accumulator.rs]: Same comment as above ;)

**Comment by jeanphilippeD** [comment]
Fixed uncomplete_events -> incomplete_events

**Pull request merged**